pkg/controller/job/job_controller.go

```diff
@@ -1361,18 +1361,33 @@ func jobSuspended(job *batch.Job) bool {
 	return job.Spec.Suspend != nil && *job.Spec.Suspend
 }
 
+// Wait for Pods to terminate
+func waitForPodsToTerminate(job *batch.Job) bool {
+	if job.Spec.PodFailurePolicy == nil {
+		return false
+	}
+	return job.Spec.PodFailurePolicy.DelayPodRecreationUntilTerminal != nil && *job.Spec.PodFailurePolicy.DelayPodRecreationUntilTerminal
```

Comment:
you can simplify using `pointer.BoolDeref(job.Spec.PodFailurePolicy.DelayPodRecreationUntilTerminal, false)`

Reply:
Nice call out!

Reply:
Updated.